lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -0,0 +1,182 @@
+;(function (define, undefined) {
+    'use strict';
+    define([
+        'gettext', 'jquery', 'underscore', 'backbone', 'js/mustache', 'js/views/fields'
+    ], function (gettext, $, _, Backbone, RequireMustache, FieldViews) {
+
+        var AccountSettingsFieldViews = {};
+
+        AccountSettingsFieldViews.EmailFieldView = FieldViews.TextFieldView.extend({
+
+            successMessage: function() {
+                return this.indicators.success + interpolate_text(
+                    gettext(
+                        'We\'ve sent a confirmation message to {new_email_address}. ' +
+                        'Click the link in the message to update your email address.'
+                    ),
+                    {'new_email_address': this.fieldValue()}
+                );
+            }
+        });
+
+        AccountSettingsFieldViews.LanguagePreferenceFieldView = FieldViews.DropdownFieldView.extend({
+
+            saveSucceeded: function () {
+                var data = {
+                    'language': this.modelValue()
+                };
+
+                var view = this;
+                $.ajax({
+                    type: 'POST',
+                    url: '/i18n/setlang/',
+                    data: data,
+                    dataType: 'html',
+                    success: function () {
+                        view.showSuccessMessage();
+                    },
+                    error: function () {
+                        view.message(
+                            view.indicators.error +
+                            gettext(
+                                'You must sign out of edX and sign back in before your language ' +
+                                'changes take effect.'
+                            )
+                        );
+                    }
+                });
+            }
+
+        });
+
+        AccountSettingsFieldViews.PasswordFieldView = FieldViews.LinkFieldView.extend({
+
+            initialize: function (options) {
+                this._super(options);
+                _.bindAll(this, 'resetPassword');
+            },
+
+            linkClicked: function (event) {
+                event.preventDefault();
+                this.resetPassword(event);
+            },
+
+            resetPassword: function () {
+                var data = {};
+                data[this.options.emailAttribute] = this.model.get(this.options.emailAttribute);
```

Comment:
Note that in the following case, password reset does not work because the email value is wrong (we may not care about this, but it confused me). FYI, @explorerleslie 

1) Initiate email change on the account settings page, but don't complete the change (so the email field shows a value that is not your current address)

2) Click reset password

The reset fails because the address that is sent is not the one actually associated with the account (with no indication of the failure).

Note that if the user has specified an email belonging to someone else, that person does NOT receive a password change email because the model appears to keep the old value since validation failed.